stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/AttachmentsPickerTabFactoryFilter.kt

```diff
@@ -47,11 +52,23 @@ internal class AttachmentsPickerTabFactoryFilter {
     private fun adjustSystemFactory(
         factory: AttachmentsPickerSystemTabFactory,
         channel: Channel,
+        messageMode: MessageMode,
     ): AttachmentsPickerSystemTabFactory {
-        // Adjust pollEnabled based on the channel config
+        // Adjust pollAllowed based on the:
+        // 1. Current message mode (only in Normal mode)
+        // 2. Channel config (are polls enabled for the channel and the user has rights)
+        // 3. Factory config (is the polls tab enabled in the factory config)
         val config = factory.config.copy(
-            pollAllowed = channel.isPollEnabled() && factory.config.pollAllowed,
+            pollAllowed = messageMode is MessageMode.Normal && channel.isPollEnabled() && factory.config.pollAllowed,
         )
         return AttachmentsPickerSystemTabFactory(config)
     }
+
+    private fun AttachmentsPickerTabFactory.isEnabledForMessageMode(mode: MessageMode): Boolean {
+        return when (this) {
+            // The default "Polls" tab is only shown in Normal mode
+            is AttachmentsPickerPollTabFactory -> mode is MessageMode.Normal
+            else -> true
+        }
+    }
```

Comment:
Wouldn't it make more sense to include this logic in the `isPickerTabEnabled` method? With this PR it makes the showing logic even more scattered across the code base